subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/InstantExecutionReport.kt

```diff
@@ -119,45 +162,52 @@ class InstantExecutionReport(
 
     private
     fun errors() =
-        problems.filterIsInstance<PropertyProblem.Error>()
+        problems.toList().filterIsInstance<PropertyProblem.Error>()
 
     private
-    fun logSummary() {
-        logger.warn(summary())
+    fun logSummary(outputDirectory: File) {
+        logger.warn(summary(outputDirectory))
     }
 
     private
-    fun writeReportFiles() {
-        outputDirectory.mkdirs()
-        copyReportResources()
-        writeJsReportData()
+    fun writeReportFiles(outputDirectory: File) {
+        require(outputDirectory.deleteRecursively()) {
```

Comment:
❓ why is it important to delete all files from the outputDirectory here? isn't that a new directory per report?

Reply:
Good catch, removed